drivers/memc/memc_renesas_ra_sdram.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_ra_sdram
+
+#include <zephyr/device.h>
+#include <sdram.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(memc_renesas_ra_sdram, CONFIG_MEMC_LOG_LEVEL);
+
+struct memc_renesas_ra_sdram_config {
+	const struct pinctrl_dev_config *pincfg;
+};
+
+static int renesas_ra_sdram_init(const struct device *dev)
+{
+	const struct memc_renesas_ra_sdram_config *config = dev->config;
+	int err;
+
+	err = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
+	if (err) {
+		LOG_ERR("pin function initial failed");
+		return err;
+	}
+	sdram_init();
```

Comment:
Current implementation just supports for SDRAM on EK-RA8D1

Reply:
Ok- I would suggest naming the driver based on the SDRAM part number then. Otherwise, users may incorrectly assume that the driver will work for any SDRAM. Another (better) option would be to implement a MEMC driver capable of configuring the SDRAM controller based on settings in devicetree, so it can work for any SDRAM chip

Reply:
I'm still concerned about the naming of this driver. This implies the SDRAM memc driver will enable SDRAM support on any board with an RA SOC, right?

What will happen when a customer designs a board with a different SDRAM part, and tries to use this driver?

One option to explore would be making the SDRAM init part of a board specific C init file.